### PR TITLE
Update MainActivity.kt

### PR DESCRIPTION
**Pull request by Dawid-Witkowski**
Updated the project to work even in 2022

### CODE REVIEW

**Comment by Dawid-Witkowski** [comment]
To clarify:
If you (in the current state) clone the repo, it will show an error in line 23 of MainActivity.kt:
image
I guess the .replace() function has changed during the years, and now, in order to make the project run you have to change the .replace() parameters to:
container ID, ListFragment instance and a tag

(should've described the problem at merge, mb)

File: app/src/main/java/ca/nick/multimodulenavigation/MainActivity.kt [hunk 1]
```diff
@@ -17,8 +17,15 @@ class MainActivity : AppCompatActivity() {
...
             supportFragmentManager.beginTransaction()
-                .replace(R.id.container, ListFragment::class.java, null, "list")
+                .replace(
+                    R.id.container,
+                    ListFragment(
+                        onItemClicked = OnItemClickedDirections(supportFragmentManager),
+                        SomeHeavyWeightDependency()
+                    ),
+                    "list"
+                )
```
Comment (nihk) [inline app/src/main/java/ca/nick/multimodulenavigation/MainActivity.kt]: instantiating a ListFragment directly will crash on process recreation, it has to go thru the FragmentFactory. this new API can be used instead.

```suggestion
                .replace<ListFragment>(R.id.container, "list")
```

**Comment by nihk** [comment]
See also: 31a95ed20d9785b7f6b1eca51a594118349aab60

**Comment by nihk** [comment]
i just did a few force pushes to master to fix a few things; should be good now.

thanks for the heads up!

**Comment by Dawid-Witkowski** [comment]
No worries, this project helped me a lot, so had to give something back 😉

**Pull request merged**